src/sentry/eventstore/base.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import absolute_import
+
+from enum import Enum
+
+from sentry import nodestore
+from sentry.utils.services import Service
+
+
+class Columns(Enum):
+    EVENT_ID = 'event_id'
+    GROUP_ID = 'group_id'
+    PROJECT_ID = 'project_id'
+    TIMESTAMP = 'timestamp'
+    CULPRIT = 'culprit'
+    LOCATION = 'location'
+    MESSAGE = 'message'
+    PLATFORM = 'platform'
+    TITLE = 'title'
+    TYPE = 'type'
+    TAGS_KEY = 'tags.key'
+    TAGS_VALUE = 'tags.value'
+    EMAIL = 'email'
+    IP_ADDRESS = 'ip_address'
+    USER_ID = 'user_id'
+    USERNAME = 'username'
+
+
+class EventStorage(Service):
+    __all__ = (
+        'minimal_columns',
+        'full_columns',
+        'get_event_by_id',
+        'get_events',
+        'get_prev_event_id',
+        'get_next_event_id',
+        'bind_nodes',
+    )
+
+    # The minimal list of columns we need to get from snuba to bootstrap an
+    # event. If the client is planning on loading the entire event body from
+    # nodestore anyway, we may as well only fetch the minimum from snuba to
+    # avoid duplicated work.
+    minimal_columns = [
+        Columns.EVENT_ID,
+        Columns.GROUP_ID,
+        Columns.PROJECT_ID,
+        Columns.TIMESTAMP,
+    ]
+
+    # A list of all useful columns we can get from snuba.
+    full_columns = minimal_columns + [
+        Columns.CULPRIT,
+        Columns.LOCATION,
+        Columns.MESSAGE,
+        Columns.PLATFORM,
+        Columns.TITLE,
+        Columns.TYPE,
+
+        # Required to provide snuba-only tags
+        Columns.TAGS_KEY,
+        Columns.TAGS_VALUE,
+
+        # Required to provide snuba-only 'user' interface
+        Columns.EMAIL,
+        Columns.IP_ADDRESS,
+        Columns.USER_ID,
+        Columns.USERNAME,
+    ]
+
+    def get_event_by_id(self, project_id, event_id, additional_columns):
+        """
+        Gets a single event given a project_id and event_id.
+
+        Keyword arguments:
+        project_id (int): Project ID
+        event_id (str): Event ID
+        additional_columns: (List[Column]) - List of addition columns to fetch - default None
+        """
+        raise NotImplementedError
+
+    def get_events(self, start, end, additional_columns,
+                   conditions, filter_keys, orderby, limit, offset):
+        """
+        Fetches a list of events given a set of criteria.
+
+        Keyword arguments:
+        start (DateTime): Start datetime - default datetime.utcfromtimestamp(0)
+        end (DateTime): End datetime - default datetime.utcnow()
+        additional_columns (List[Column]): List of additional columns to fetch - default None
+        conditions (List[List[str, str, Any]]): List of conditions to fetch - default None
```

Comment:
nit: If you use the mypy typing syntax, please use Sequence instead of List and Mapping instead of Dict. This is the abstract version of what you need.